include/range/v3/utility/concepts.hpp

```diff
@@ -655,8 +626,13 @@ namespace ranges
         template<typename T, typename...Us>
         using Constructible = concepts::models<concepts::Constructible, T, Us...>;
 
-        template<typename T>
-        using DefaultConstructible = concepts::models<concepts::DefaultConstructible, T>;
+        template<typename T, typename...Us>
+        using ConstructibleRef =
+            meta::apply<
+                meta::if_<std::is_reference<T>,
+                    meta::quote<std::is_constructible>,
+                    meta::quote<Constructible>>,
+                T, Us...>;
```

Comment:
> And should Constructible subsume Initializable? In this pull request, ConstructibleRef is a bit of a bolt-on afterthought.

In Concepts, an equivalent definition would be:
```
template <class T, class...Args>
concept bool ConstructibleRef =
  Constructible<T, Args...> ||
  (std::is_reference<T>::value && requires(Args&&...args) {
    T(forward<Args>(args)...);
  });
```
from which it is clear that Constructible<T, Args...> subsumes ConstructibleRef<T, Args...>. The formulation I used for range-v3 performs the conditional early to avoid instantiating the failing branch for "performance" and to avoid a hard error in an earlier version of Destructible that wasn't guarded with `meta::if_<std::is_object<T>>`.